### DISCUSSION TITLE
How to get the number of active rooms in the current thread?

### DISCUSSION THREAD

**Discussion by kelvinlee** [colyseus/colyseus#605] [Q&A]
I need to check the active rooms, when I reload the code.

**Comment by endel** [comment]
You can use [`matchMaker.query()`](https://docs.colyseus.io/server/matchmaker/#queryconditions) to fetch all active rooms, loop through them while checking which entry matches the local `processId` (`matchMaker.processId`)

```typescript
import { matchMaker } from "colyseus";

const roomsAvailable = await matchMaker.query({});

roomsAvailable.forEach((roomAvailable) => {
  if (roomsAvailable.processId === matchMaker.processId) {
    console.log(roomsAvailable, "on current process");
  }
})
```

Hope this helps
